build-tools/packages/build-cli/src/commands/merge/branches.ts

```diff
@@ -54,6 +55,48 @@ export default class MergeBranch extends BaseCommand<typeof MergeBranch> {
 		...BaseCommand.flags,
 	};
 
+	// `findConflictIndex` return the index of the conflicting commit. If there are no conflicts found, then it would return the last index
+	async findConflictIndex(commitIds: string[], gitRepo: GitRepo): Promise<number> {
+		const length = commitIds.length;
+
+		if (length === 0) {
+			return -1;
+		}
+
+		// Check the first commit separately
+		const firstCommit = commitIds[0];
+		const firstResponse = await gitRepo.canMergeWithoutConflicts(firstCommit);
+		this.log(`Response from merge check for ${commitIds[0]}: ${firstResponse}`);
+		if (firstResponse === false) {
+			return 0;
+		}
+
+		// Check the rest of the commitIds
+		for (let i = 1; i < length; i++) {
+			const commit = commitIds[i];
+			// eslint-disable-next-line no-await-in-loop
+			const response = await gitRepo.canMergeWithoutConflicts(commit);
+			this.log(`Response from merge check for ${commitIds[i]}: ${response}`);
+			if (response === false) {
+				return i;
+			}
+		}
+
+		return length - 1; // No conflicts found, return the last index
+	}
+
+	/**
+	 * `cherryPickAllCommits` function will merge all commitIds in next branch
+	 * we cannot use gitRepo.merge as it would create a merge commit for individual commit.
+	 */
+	async cherryPickAllCommits(commitIds: string[], gitRepo: GitRepo): Promise<void> {
+		for (const commit of commitIds) {
+			// eslint-disable-next-line no-await-in-loop
+			const response = await gitRepo.cherryPick(commit);
+			this.log(`Merging commit ${commit}: ${response}`);
+		}
+	}
```

Comment:
We definitely don't want to cherry-pick, because that will create duplicate commits with confusing history. I think you should merge the most recent commit in the list that didn't conflict. That will create one merge commit, and then you open the PR from that merge commit to next. Then, when merging, instead of using the GitHub UI the person doing the merge fast-forwards next to the merge commit and pushes.